specs/flyout.md

```diff
@@ -0,0 +1,152 @@
+
+### Background
+A Flyout can be less intrusive to the user experience than a Dialog and should be used instead of a Dialog when information or input is needed to be displayed, but not in the center of the screen, such as in the case of collection of additional info before an action, displaying information that is only relevant some of the time, or displaying more information.
+
+### Use Cases
+- As a Dropdown for a menu, navigation bar, etc.
+- In a context menu as the secondary menu/action.
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyouts anchor element. This way it can be dynamic and adapt to the available space if `responsive` is true for the Flyout.
+
+**Light Dismiss (aka Soft Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key.
+
+**Overlay:** - A control of whether or not the Flyout has an overlay effect that dims the surrounding UI when the Flyout is active.
+
+### Prior Art/Examples
+##### Context Menu
+*Note:* the "Manage, Apply, Show more options" is the Flyout for this example.
+![Context Menu](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-nested.png)
+
+##### Text Only
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+##### Actionable
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `anchor` - string, the html id of the HTMLElement that the Flyout is attached to
+- `overlay` - boolean, specifies whether an overlay is drawn to dim the out of scope UI until the Flyout is dismissed
+- `light-dismiss` - boolean, specifies whether or not the flyout can be dismissed by clicking elsewhere or pressing `esc` key, defaults to true
+- `visible` - boolean, whether or not the Flyout is visible, defaults to undefined, as this is controlled by the author
+- `responsive` - boolean, whether or not the positioning is responsive based on available space, defaults to true, uses Anchored Region logic
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic
+    - top-centered
+    - right-centered
+    - bottom-centered
+    - left-centered
+    - top-left-corner
+    - top-right-corner
+    - bottom-right-corner
+    - bottom-left-corner
+    - top-left-aligned
+    - right-top-aligned
+    - bottom-left-aligned
+    - left-top-aligned
+        - **ASK (1)**
```

Comment:
top-right-aligned, bottom-right-aligned and others may be missing or I may not be clear on what "aligned" means?  I'm thinking it controls which edge the flyout abuts to and the direction it grows?

Reply:
I should make a diagram for this...
I was thinking `top` meant the general location of the flyout is *above* its anchor element, and `right-aligned` meant the *left* edge of the flyout would be aligned with the *right* edge of the anchor. That does seem confusing as I write it out here. I'll try to clarify this, and think of better wording.